CMakeLists.txt

```diff
@@ -47,6 +47,27 @@ set(H3_SOVERSION 1)
 
 project(h3 VERSION ${H3_VERSION})
 
+set(H3_FLAGS "")
+if(NOT WIN32)
+    # Compiler options are set only on non-Windows, since these options
+    # are not correct for MSVC.
+    string(CONCAT H3_FLAGS ${H3_FLAGS} "-Wall ")
+
+    if(CMAKE_BUILD_TYPE STREQUAL "Debug")
```

Comment:
So I once thought this was a good idea too. Turns out, checking for "Debug" build might be done at build time instead of configuration time, so you need to use a generator expression here. Something like this should do it:

```
string(CONCAT H3_FLAGS "$<$<CONFIG:Debug>:-g -gdwarf-2 -g3 -O0 -fno-inline -fno-eliminate-unused-debug-types> ")
```

More on this topic here: https://stackoverflow.com/a/24470998/1930331.

For the record, I'm not convinced it's a good idea to use `-O0` or `-g` in these options because CMake will add them as the default debug flags.

Reply:
To be clear, please remove the if statement entirely and you can append this generator expression to the earlier `-Wall` if you'd like.